### PR TITLE
Adding functionality to add version even if base version is empty

### LINKED ISSUE #1: Unable to add new version
**Steps to reproduce**
- Click on 'Add a new version'
- Enter the details in 'Version id' and 'Upstream version'. I kept 'Base Version' as empty and expected to use the latest one by default
- Click on 'Add version'
- It will show a dialogue box with an error

Below is the error screenshot :
image

### PR DESCRIPTION
**Pull request by harshmadhani**
Fixes https://github.com/harshmadhani/quarkus-prod/issues/1

### CODE REVIEW

**Comment by harshmadhani** [comment]
@michalszynkiewicz please do review and merge

File: src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ public void updateExtension(@PathParam String version, @PathParam String extensi
 
     @POST
     public void addVersion(Version version) {
+        if (version.getBaseVersion().isEmpty()){
+            double baseVersion = getVersions().stream().mapToDouble(v->Double.parseDouble(v.getId()))
```
Comment (michalszynkiewicz) [inline src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java]: this should be handled by selecting the newest/default version in `toRealVersion`

Comment (harshmadhani) [reply src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java]: Ok, but there is no check/validation to ensure that it happens every time we add a new version. If somebody adds a new version without checking `newest`, then it pass a null `Version` object. We can change the above code to a simple `version.setBaseVersion(MockStorage.defaultVersion);`

Comment (michalszynkiewicz) [reply src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java]: I insist on using toRealVersion here ;) UI can send `newest` when user doesn't provide a version number

Comment (harshmadhani) [reply src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java]: Okay,good idea !

Comment (harshmadhani) [reply src/main/java/com/redhat/quarkus/prodapp/ProductVersionsResource.java]: Done !

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**